src/build/cargo.rs

```diff
@@ -367,25 +354,22 @@ impl Executor for RlsExecutor {
             // so the dep-info is ready by the time we return from this callback.
             // NB: In `workspace_mode` regular compilation is performed here (and we don't
             // only calculate dep-info) so it should fix the problem mentioned above.
-            // Since ProcessBuilder doesn't allow to modify args, we need to create
-            // our own command here from scratch here.
-            for a in &args {
-                // Emitting only dep-info is possible only for final crate type, as
-                // as others may emit required metadata for dependent crate types
-                if a.starts_with("--emit") && is_final_crate_type && !self.workspace_mode {
-                    cmd.arg("--emit=dep-info");
-                } else if a != "-Zsave-analysis" {
```

Comment:
Does this mean we don't ever pass `"-Zsave-analysis"` to the deps?

Reply:
Sorry, to be clear, when running on the deps, we use the RLS as a rustc shim which effectively always has -Zsave-analysis set. So we never explicitly set -Zsave-analysis - for black-listed deps, we don't want it, for non-black-listed-deps we use the RLS as a shim, and for the primary crates we call rustc directly (where again collecting save-analysis data does not require the flag).